kin-sdk/AccountInfoRetriver.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Threading.Tasks;
+using kin_base;
+using kin_base.responses;
+
+namespace kin_sdk
+{
+    class AccountInfoRetriver
+    {
+        private static readonly Asset KinAsset = new AssetTypeNative();
+        private readonly Server server;
+        internal AccountInfoRetriver(Server server)
+        {
+            this.server = server;
+        }
+        internal async Task<decimal> GetBalance(string publicAddress)
+        {
+            try
+            {
+                AccountResponse accountResponse = await this.server.Accounts.Account(publicAddress);
+                foreach (Balance balance in accountResponse.Balances)
+                {
+                    if (balance.Asset.Equals(KinAsset))
+                    {
+                        return Decimal.Parse(balance.BalanceString);
+                    }
+                }
+            }
+            catch  (kin_base.requests.HttpResponseException e)
+            {
+                if (e.StatusCode == 404)
+                {
+                    throw new AccountNotFoundException(publicAddress);
+                }    
+                else
+                {
+                    throw new OperationFailedException("Failed to get account balance", e);
```

Comment:
This can be thrown under the for loop and replaced with `throw;`

Reply:
Im stay close to the android-sdk, so I'd like to throw `OperationFailedException` here and not the original  `HttpResponseException`

Reply:
I wasn't suggesting that it be removed, it can put moved into the try statement